Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -700,13 +704,26 @@ class Gx2Fitter {
 
       // calculate delta params [a] * delta = b
       deltaParams = BoundVector::Zero();
-      const ActsVector<reducedMatrixSize> deltaParamsReduced =
-          aMatrix.topLeftCorner<reducedMatrixSize, reducedMatrixSize>()
-              .colPivHouseholderQr()
-              .solve(bVector.topLeftCorner<reducedMatrixSize, 1>());
-
-      for (size_t idp = 0; idp < reducedMatrixSize; idp++) {
-        deltaParams(idp, 0) = deltaParamsReduced(idp, 0);
+      if (gx2fOptions.zeroField) {
+        const size_t reducedMatrixSize = 4;
+        const ActsVector<reducedMatrixSize> deltaParamsReduced =
+            aMatrix.topLeftCorner<reducedMatrixSize, reducedMatrixSize>()
+                .colPivHouseholderQr()
+                .solve(bVector.topLeftCorner<reducedMatrixSize, 1>());
+
+        for (size_t idp = 0; idp < reducedMatrixSize; idp++) {
+          deltaParams(idp, 0) = deltaParamsReduced(idp, 0);
+        }
```

Comment:
Is this `deltaParams.topLeftCorner<reducedMatrixSize, 1>()`? If you you can skip the `for` loop and assign directly, I think.